presto-testng-services/src/main/java/io/prestosql/testng/services/ReportMultiThreadedBeforeOrAfterMethod.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.testng.services;
+
+import org.testng.IClassListener;
+import org.testng.ITestClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Method;
+
+import static com.google.common.base.Throwables.getStackTraceAsString;
+import static io.prestosql.testng.services.Listeners.reportListenerFailure;
+import static java.lang.String.format;
+
+public class ReportMultiThreadedBeforeOrAfterMethod
+        implements IClassListener
+{
+    @Override
+    public void onBeforeClass(ITestClass testClass)
+    {
+        try {
+            reportMultiThreadedBeforeOrAfterMethod(testClass);
+        }
+        catch (RuntimeException | Error e) {
+            reportListenerFailure(
+                    ReportMultiThreadedBeforeOrAfterMethod.class,
+                    "Failed to process %s: \n%s",
+                    testClass,
+                    getStackTraceAsString(e));
+        }
+    }
+
+    private void reportMultiThreadedBeforeOrAfterMethod(ITestClass testClass)
```

Comment:
Please make it `@VisibleForTesting` and please write few tests for it.

Reply:
This is much more simple than `ReportUnannotatedMethods` and i do not think we need a test for it now.

Reply:
And so it should be simple to add tests.